vignettes/design.qmd

```diff
@@ -58,4 +58,159 @@ if the processing time per file is substantial.
 ```{mermaid}
 %%| fig-alt: "A flowchart showing the data flow within `registers2parquet`, showing how SAS files are imported, potentially joined, and then exported as Parquet files with a year partitioning."
 %%| file: flow.mmd
-```
\ No newline at end of file
+```
+
+## Functions
+
+There is one main function and several helper functions that we expose
+in this package. Each function is listed below with a short description.
+
+### Main function: `convert_register_to_parquet()`
+
+This is the main function of the package. It takes as input one or more
+SAS files, reads them, merges them if necessary, and saves them as a
+Parquet file. If multiple files are given, the function looks for a year
+in the file names to use as partitioning. The function also removes any
+duplicate rows before saving the data. The output is the path to the
+created Parquet file(s) as specified by the `output_path` parameter.
+
+<!-- TODO: Add link to reference documentation when ready -->
+
+``` r
+#' Convert register SAS file(s) and save to Parquet format
+#'
+#' If multiple paths are given, the function looks for a year (4 digits) in the
+#' file names to use the year as partition, See `vignettes("design")` for more
+#' information about the partitioning.
+#'
+#' If multiple files are given with the same year, they are merged with a full join.
+#'
+#' If any duplicate rows found, they are removed before saving to Parquet.
+#'
+#' @param input_path A character vector with the path to the register SAS file(s).
+#' @param output_path The path with the directory to save the output Parquet file to.
+#'
+#' @returns Returns a character vector with the path to the created Parquet file(s)
+#'   (from `output_path`), so it can easily be used in a targets pipeline.
+#' @export
+#' @examples
+#' \dontrun{
+#' convert_register_to_parquet(c("bef2018.sas7bdat",

```

Comment:
Best to show the example using the get path helpers, rather than direct character vectors.